tests/test_main.py

```diff
@@ -339,3 +339,56 @@ def test_main_noxfile_options_sessions():
         # Verify that the config looks correct.
         config = honor_list_request.call_args[1]["global_config"]
         assert config.sessions == ["test"]
+
+
+@pytest.mark.parametrize(("isatty_value", "expected"), [(True, True), (False, False)])
+def test_main_color_from_istty(isatty_value, expected):
+    sys.argv = [sys.executable]
```

Comment:
Don't you want to explicitly tell `pytest` that you are patching `sys` so it can clean up for you? (Similar at line 370 and possibly elsewhere.)

Reply:
Done (everywhere)